### ISSUE TITLE
java.lang.NoClassDefFoundError: Failed resolution of: Lio/smooch/core/utils/StringUtils

### ISSUE THREAD

**Issue by rubnov** [zendesk/sunshine-conversations-android#142]
After updating to version 7 of Smooch Android SDK, our crashes when calling `ConversationActivity.build().show()`:

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: nl.hgrams.passenger, PID: 8932
    java.lang.NoClassDefFoundError: Failed resolution of: Lio/smooch/core/utils/StringUtils;
        at io.smooch.ui.ConversationActivityBuilder.intent(Unknown Source:9)
        at io.smooch.ui.ConversationActivityBuilder.show(Unknown Source:0)
        at nl.hgrams.passenger.PSApplicationClass$1.run(PSApplicationClass.java:231)
        at io.smooch.core.service.SmoochService$32.run(Unknown Source:23)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "io.smooch.core.utils.StringUtils" on path: DexPathList[[zip file "/data/app/nl.hgrams.passenger-k32Rv9c7j7jZuSP2sV9gag==/base.apk"],nativeLibraryDirectories=[/data/app/nl.hgrams.passenger-k32Rv9c7j7jZuSP2sV9gag==/lib/arm64, /data/app/nl.hgrams.passenger-k32Rv9c7j7jZuSP2sV9gag==/base.apk!/lib/arm64-v8a, /system/lib64, /product/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:196)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at io.smooch.ui.ConversationActivityBuilder.intent(Unknown Source:9) 
        at io.smooch.ui.ConversationActivityBuilder.show(Unknown Source:0) 
        at nl.hgrams.passenger.PSApplicationClass$1.run(PSApplicationClass.java:231) 
        at io.smooch.core.service.SmoochService$32.run(Unknown Source:23) 
        at android.os.Handler.handleCallback(Handler.java:883) 
        at android.os.Handler.dispatchMessage(Handler.java:100) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 

```

Initialization in ApplicatonClass: 

```
    public void initSmooch() {
        Settings settings = new Settings(<INTEGRATION_ID>);
        settings.setMapsApiKey(getString(R.string.maps_key));
        Smooch.init(this, settings, new SmoochCallback<InitializationStatus>() {
            @Override
            public void run(@NonNull Response<InitializationStatus> response) {
                if (response.getData() == InitializationStatus.SUCCESS) {
                    setDataForSmoochUser();
                    ConversationActivity.builder().show(getApplicationContext());  // --> CRASH
                }
                else {
                    Utils.appendLog("Error Smooch init: " + response.getData(), "E", Constants.OTHER);
                }
            }
        });
    }

```

build.gradle: 

```
    implementation 'io.smooch:core:7.+'
    implementation 'io.smooch:ui:7.+'
    implementation 'com.android.support:support-core-utils:28.0.0'
    implementation 'com.android.support:support-compat:28.0.0'
    implementation 'com.android.support:support-core-ui:28.0.0'
    implementation "com.android.support:exifinterface:28.0.0"
    implementation "com.android.support:recyclerview-v7:28.0.0"
    implementation "com.android.support:support-media-compat:28.0.0"
```

Using Android Studio 3.6.1
compileSdkVersion 29
buildToolsVersion '29.0.2'

**Comment by rubnov** [comment]
Same crash when invoking `ConversationActivity.builder().show()` from the activity itself.

**Comment by wmora** [comment]
Thanks for reporting. The team identified the problem and will release a new version on Monday. We’ll update this issue once released

**Comment by DataWearsAHood** [comment]
Hello Rotem,

Looks like this fix was [released in v7.0.1](https://github.com/zendesk/sunshine-conversations-android/releases/tag/7.0.1)!
